### PR TITLE
Reject invalid and approve SOLs

### PR DESCRIPTION
**Pull request by garious**
#### Problem

Currently, the UI allows a user to sign any message after approving its hash.

#### Proposed Changes

* Reject anything that isn't a valid Solana message (without parsing the instructions)
* If a simple system transfer, show amount of SOL being transferred

Next up, approve `from`, `to` and `fee_payer` pubkeys.

### CODE REVIEW

File: rust-tests/src/lib.rs [hunk 1]
```diff
@@ -64,6 +64,39 @@ mod tests {
         assert_eq!(pubkey_set.len(), 5); // Ensure keys at various derivation paths are unique
     }
 
+    /// This test requires interactive approval of message signing on the ledger.
+    #[test]
+    #[serial]
+    fn ledger_sign_garbage_test() {
+        let wallet_manager = initialize_wallet_manager();
```
Comment (garious) [inline rust-tests/src/lib.rs]: @CriesofCarrots, the existing code set a precedent for copy-pasta.

File: src/signMessage.c [hunk 2]
```diff
@@ -82,7 +111,14 @@ void handleSignMessage(uint8_t p1, uint8_t p2, uint8_t *dataBuffer, uint16_t dat
 
         sendResponse(0, false);
     } else {
-        ux_flow_init(0, ux_display_message, NULL);
+        if (system_transfer_err == 0) {
+	    uint32_t sol = info.lamports / 1000000000;
+	    uint32_t lamports = info.lamports % 1000000000;
+            snprintf(messageTransfer, MESSAGE_TRANSFER_SIZE, "%u, %u", sol, lamports);
```
Comment (garious) [inline src/signMessage.c]: This is super ugly. We need a library function for printing a decimal amount of SOL.

File: libsol/system_instruction.c [hunk 3]
```diff
@@ -59,7 +40,7 @@ int parse_system_transfer_instructions(Parser* parser, MessageHeader* header, Sy
     Pubkey system_program_id = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
     BAIL_IF(memcmp(program_id, &system_program_id, PUBKEY_SIZE));
 
-    BAIL_IF(parse_system_transfer_instruction(&instruction, header->pubkeys, header->pubkeys_length, info));
+    BAIL_IF(parse_system_transfer_instruction(&instruction, header->pubkeys, header->pubkeys_header.pubkeys_length, info));
```
Comment (garious) [inline libsol/system_instruction.c]: @t-nelson, for the curious, here was the bug: I moved `pubkeys_length` to a different struct and forgot to delete the original, leaving an uninitialized one and one spot where the code referenced it.

File: rust-tests/src/lib.rs (hunk 1 above)
Comment (CriesofCarrots) [reply rust-tests/src/lib.rs]: I'll rework that in a helper fn at some point.

**Pull request merged**